From: Don Dugger <ddugger@redhat.com>
Date: Thu, 2 Sep 2010 20:43:33 -0400
Subject: [xen] oprofile: add support for Core i7 and Atom
Message-id: <201009022043.o82KhXxO005684@sobek.n0ano.com>
Patchwork-id: 28030
O-Subject: [RHEL 5.6 PATCH 1/3] BZ 538564: xenoprof: add support for Core i7 and
	Atom.
Bugzilla: 538564
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>

Signed-off-by: Yang Zhang <yang.zhang@intel.com>
Signed-off-by: Yang Xiaowei <xiaowei.yang@intel.com>

Upstream status: CS 65968459de75

Signed-off-by: Don Dugger <donald.d.dugger@intel.com>
---
 arch/x86/oprofile/nmi_int.c |   12 +++++++-----
 1 files changed, 7 insertions(+), 5 deletions(-)

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/oprofile/nmi_int.c b/arch/x86/oprofile/nmi_int.c
index 595e73f..57047f3 100644
--- a/arch/x86/oprofile/nmi_int.c
+++ b/arch/x86/oprofile/nmi_int.c
@@ -303,22 +303,24 @@ static int __init ppro_init(char ** cpu_type)
 		*cpu_type = "i386/pii";
 		break;
 	case 6 ... 8:
+	case 10 ... 11:
 		*cpu_type = "i386/piii";
 		break;
 	case 9:
+	case 13:
 		*cpu_type = "i386/p6_mobile";
 		break;
-	case 10 ... 13:
-		*cpu_type = "i386/p6";
-		break;
 	case 14:
 		*cpu_type = "i386/core";
 		break;
-	case 15: case 23:
+	case 15: case 23: case 29:
 		*cpu_type = "i386/core_2";
 		break;
 	case 26:
-		*cpu_type = "i386/core_2";
+		*cpu_type = "i386/core_i7";
+		break;
+	case 28:
+		*cpu_type = "i386/atom";
 		break;
 	default:
 		/* Unknown */
